### PR TITLE
Fix Python 3.11 compatibility

### PR DESCRIPTION
**Pull request by roehling**

### CODE REVIEW

**Review by kozlovsky — CHANGES_REQUESTED** [review]
Looks impressive! Thank you, guys, for this great PR!

We need to make sure that the code works for all supported Python versions. Currently, the tests are broken on Python 3.8 because of a non-supported `indent` argument of `ast.dump()`. Can we ask you to make the code with Python >= 3.6?

As two possible ways to fix it, it is possible to monkey-patch `ast.dump` for the previous Pythons to support the `indent` argument or make the indent keyword argument optional and de-denting the `expected` ast representation for previous Pythons.

Other than that, it all looks great!

**Comment by jspricke** [comment]
I opted for skipping the test in old Python versions. Both fixing the indention or dedent to match the output seems to need non trivial amount of code.

**Comment by boxed** [comment]
What's the status here?

If just some indent is changed but the code works, might I suggest duplicating the tests and set skip properly for each version?

File: pony/orm/tests/test_decompiler.py [hunk 1]
```diff
@@ -93,7 +96,85 @@ def get_condition_values(cond):
...
+            return
+
+        code = compile(src, '<?>', 'eval').co_consts[0]
+        import dis
+        print(dis.dis(code))
+        dc = Decompiler(code)
+        expected = textwrap.dedent(expected).strip()
+        self.maxDiff = None
+        self.assertMultiLineEqual(expected, ast.dump(dc.ast, indent=2))
+
```
Comment (frodo821) [inline pony/orm/tests/test_decompiler.py:111]: I'm suggesting a not-so-better workaround:

```suggestion
        self.assertMultiLineEqual(self._strip_each_line(expected), ast.dump(dc.ast, indent=2))

    def _strip_each_line(self, string: str) -> str:
        return ' '.join(
            l.strip() for l in string.splitlines()
        ).replace('( ', '(').replace('[ ', '[').strip()
```

As far as I tested, it looks like it is working.

**Comment by pjdewitte** [comment]
It is great to see work is being done to support Python 3.11!

I tested this pull request and found an expression that no longer decompiles properly:

`(person for person in Person if (person.address is None or person.address.street == "void"))`

When using Python 3.11, "is None" decompiles to "is not None". When using Python 3.10, it does work correctly.

I dug into the decompiling and found that `POP_JUMP_FORWARD_IF_NONE` gets negated, while it shouldn't in this case. 

BTW: `assertDecompilesTo` is great for checking this. The "|not-so-better workaround" seems fine to me: stripping whitespace makes sense.

**Comment by roehling** [comment]
> When using Python 3.11, "is None" decompiles to "is not None". When using Python 3.10, it 
> I dug into the decompiling and found that `POP_JUMP_FORWARD_IF_NONE` gets negated, while it shouldn't in this case. 

You are right! I remember I found this before and thought I had fixed this, but apparently I forgot to update the PR accordingly. Sorry!

**Comment by pjdewitte** [comment]
@roehling Thanks for fixing this! 

My testing didn't reveal other bugs.

**Comment by nburns** [comment]
really looking forward to using this

**Comment by fredrik-smedberg** [comment]
@kozlovsky please check.

**Comment by mmuehle** [comment]
Maybe it would be a good idea to drop support of deprecated python version 3.6 and 3.7 for future releases? No one should use these versions in production code because they have no security patches any more.

Peoples which like to have support for older version can stull use previous releases of pony. 

My team would like to use python 3.11 as it is the actual python version with active support. Therfore it would be great to merge these changes into a new official release.

**Comment by kozlovsky** [comment]
Thank you for this PR, and I am sorry for the delay! Confirmed it works for Python 3.8-3.11. I agree that it is time to drop support of Python versions < 3.8.

**Comment by boxed** [comment]
Could we get a release on pypi for this? I would love to be able to close the open issue I have on mutmut for this..

**Comment by nburns** [comment]
Seems to not support `JUMP_BACKWARD` correctly?

On the latest version I get the following error:
```
models/permissions/position.py:58: in positions_with_permission_type
    empty_set = select(p for p in Position if False)
../../.asdf/installs/python/3.11.5/lib/python3.11/site-packages/pony/orm/core.py:5560: in select
    return make_query(args, frame_depth=cut_traceback_depth+1)
../../.asdf/installs/python/3.11.5/lib/python3.11/site-packages/pony/orm/core.py:5546: in make_query
    tree, external_names, cells = decompile(gen)
../../.asdf/installs/python/3.11.5/lib/python3.11/site-packages/pony/orm/decompiling.py:43: in decompile
    decompiler = Decompiler(codeobject)
../../.asdf/installs/python/3.11.5/lib/python3.11/site-packages/pony/orm/decompiling.py:162: in __init__
    decompiler.decompile()
../../.asdf/installs/python/3.11.5/lib/python3.11/site-packages/pony/orm/decompiling.py:273: in decompile
    throw(DecompileError('Unsupported operation: %s' % opname))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

exc_type = DecompileError('Unsupported operation: JUMP_BACKWARD'), args = (), kwargs = {}

    def throw(exc_type, *args, **kwargs):
        if isinstance(exc_type, Exception):
            assert not args and not kwargs
            exc = exc_type
        else: exc = exc_type(*args, **kwargs)
        exc.__cause__ = None
        try:
            if not (pony.MODE == 'INTERACTIVE' and options.CUT_TRACEBACK):
>               raise exc
E               pony.orm.decompiling.DecompileError: Unsupported operation: JUMP_BACKWARD

../../.asdf/installs/python/3.11.5/lib/python3.11/site-packages/pony/utils/utils.py:99: DecompileError
```

**Pull request merged**